tests/test_drop_counters.py

```diff
@@ -0,0 +1,526 @@
+
+        # cleanup
+        self.delete_drop_counter(name)
+        self.remove_drop_reason(name, reason1)
+
+    def test_addDropReasonMultipleTimes(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        asic_state_table = swsscommon.Table(self.asic_db, ASIC_STATE_TABLE)
+        flex_counter_table = swsscommon.Table(self.flex_db, FLEX_COUNTER_TABLE)
+
+        name = 'ADD_TEST'
+        reason1 = 'L3_ANY'
+
+        self.create_drop_counter(name, SWITCH_INGRESS_DROPS)
+        self.add_drop_reason(name, reason1)
+        time.sleep(3)
+
+        # Quick sanity check to ensure the counter has been created
+        assert len(asic_state_table.getKeys()) == 1
+        assert len(flex_counter_table.getKeys()) > 0
+
+        reason2 = 'ACL_ANY'
+        self.add_drop_reason(name, reason2)
+        time.sleep(3)
+
+        self.add_drop_reason(name, reason2)
+        time.sleep(3)
+
+        # Check that the ASIC state is correct
+        asic_keys = asic_state_table.getKeys()
+        assert len(asic_keys) == 1
+        assert self.checkASICState(asic_keys[0], ASIC_COUNTER_SWITCH_IN_TYPE, [reason1, reason2])
+
+        # cleanup
+        self.delete_drop_counter(name)
+        self.remove_drop_reason(name, reason1)
+        self.remove_drop_reason(name, reason2)
+
+    def test_addInvalidDropReason(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        asic_state_table = swsscommon.Table(self.asic_db, ASIC_STATE_TABLE)
+        flex_counter_table = swsscommon.Table(self.flex_db, FLEX_COUNTER_TABLE)
+
+        name = 'ADD_TEST'
+        reason1 = 'L3_ANY'
+
+        self.create_drop_counter(name, SWITCH_INGRESS_DROPS)
+        self.add_drop_reason(name, reason1)
+        time.sleep(3)
+
+        # Quick sanity check to ensure the counter has been created
+        assert len(asic_state_table.getKeys()) == 1
+        assert len(flex_counter_table.getKeys()) > 0
+
+        dummy_reason = 'ZOBOOMBAFOO'
+        self.add_drop_reason(name, dummy_reason)
+        time.sleep(3)
+
+        reason2 = 'ACL_ANY'
+        self.add_drop_reason(name, reason2)
+        time.sleep(3)
+
+        # Check that the ASIC state is correct
+        asic_keys = asic_state_table.getKeys()
+        assert len(asic_keys) == 1
+        assert self.checkASICState(asic_keys[0], ASIC_COUNTER_SWITCH_IN_TYPE, [reason1, reason2])
+
+        # cleanup
+        self.delete_drop_counter(name)
+        self.remove_drop_reason(name, reason1)
+        self.remove_drop_reason(name, reason2)
+
+    def test_removeDropReasonMultipleTimes(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        asic_state_table = swsscommon.Table(self.asic_db, ASIC_STATE_TABLE)
+        flex_counter_table = swsscommon.Table(self.flex_db, FLEX_COUNTER_TABLE)
+
+        name = 'ADD_TEST'
+        reason1 = 'L3_ANY'
+        reason2 = 'ACL_ANY'
+
+        self.create_drop_counter(name, SWITCH_INGRESS_DROPS)
+        self.add_drop_reason(name, reason1)
+        self.add_drop_reason(name, reason2)
```

Comment:
Do you expect reason2 to be created? Some comments in your test flow would help a lot.